src/cigar.py

```diff
@@ -65,4 +74,29 @@ def edits_to_cigar(edits: str) -> str:
     '1M1D6M1I4M'
 
     """
-    return ""
+
+    cigar = ""
+
+    for i in range(len(edits)):
+
+        if i == 0:
+            # Starts a count of the repeating characters with the first character as a reference.
+            character = edits[i]
+            count = 1
+
+        else:
+
+            if edits[i] == character:
+                # Continues the count if the next character is a repeating character.
+                count += 1
+
+            else:
+                # Appends the corresponding count and character pair to the cigar string
+                cigar += f'{count}{character}'
+                # Starts a new count of the repeating characters with the current character as a reference.
+                character = edits[i]
+                count = 1
+                
+    cigar += f'{count}{character}'
+
+    return cigar
```

Comment:
Fra hvad jeg lige kan se og teste mig frem til, så er problemet i den her funktion - den tillader nemlig ikke at `edits_to_cigar` får en tom string som input

Reply:
Jep. Det er linjen

```python
cigar += f'{count}{character}'
```

der skaber baladen.

Starter man med en top streng, så bliver variablene `count` og `character` aldrig initialiseret, så så brokker Python sig.

Prøv at lave en invariant for løkken, således at hvis `cigar` er tom før I går ind i løkken så er invarianten opfyldt, og hvis I forlader løkken så har I det endelige resultat. Jeg ville også anbefale at tage et kik på `split_blocks()` funktionen--den kan være nyttig her.